workspaces/ballerina/ballerina-extension/CHANGELOG.md

```diff
@@ -2,6 +2,53 @@
 
 All notable changes to the "Ballerina" extension will be documented in this file.
 
+## **5.2.0** (2025-07-12)
+
+### Major Features
+
+- **Bundled Language Server**: Ballerina Language Server is now bundled with the extension, eliminating separate installation requirements and improving startup performance
+- **Configurable Editor v2**: Complete redesign of the configuration editor with enhanced UI/UX and improved functionality
+- **Type Editor Revamp**: A redesign of the type editor to improve feature discoverability and deliver a better user experience
+
+### Added
+
+- Enhanced AI file upload support with additional file types for improved analysis capabilities
+- Documentation display in Signature Help for a better developer experience during code completion
+- Enhanced service resource creation with comprehensive validation system for base paths, resource action calls, reserved keywords, and new UX for creating HTTP responses
+
+### Changed
+
+- **Project Management**: Refactored artifacts management and navigation system for better organization
```

Comment:
```suggestion
- **Integration Management**: Refactored artifacts management and navigation
```
seems the same point is duplicated under **User Interface**: , shall we remove that (L50)